src/posts/about-pair-programming/index.md

```diff
@@ -0,0 +1,59 @@
+---
+title: "About pair programming"
+description: 
+date: 2024-02-21
+thumb: "bonnie-kittle-MUcxe_wDurE-unsplash.jpg"
+thumbCaption: "Photo by Bonnie Kittle via Unsplash"
+thumbCaptionHref: "https://unsplash.com/photos/two-guinea-pigs-eating-carrot-MUcxe_wDurE"
+tags:
+  - pair programming
+  - development
+  - methodology
+  - inclusivity
+---
+When I first heard about pair programming I didn't like the concept.
+I'm much faster on my own without someone else slowing me down.
+
+And I was right - at the time.
+
+This was in university and since I had programmed a lot before I started at university I solved the assignments quickly.
+Working together with someone who was learning programming would indeed slow me down,
+and slowing down would mean less time to play video games or hang out with my friends.
+So I worked alone as much as possible.
+
+Nowadays,
+I no longer mostly implement bubble sort, linked lists or rock-paper-scissor simulators but real software with complex requirements, large codebases and actual users.
+The focus is no longer on passing the assignment as quickly as possible but to deliver qualitative software that provides value to my customer
+(and likely to their customers as well).
+
+Pair-programming has several benefits.
+Chief among them being that code is being collaboratively developed.
+This means that less bugs enter the code,
+you get stuck less by helping each other think,
+and both code and design is being reviewed during implementation instead of after.
+This continuous review also means that you can skip the PR review step and push straight to main once any pipelines pass.
```

Comment:
Jag håller inte helt med där. Ett par "oförstörda" ögon är inte dumt, i min erfarenhet.

Reply:
Och det ställer krav på vem som parprogrammerar. I ett projekt har vi ett par juniorer - om de parprogrammerar en PR så kan inte den gå in utan granskning.

Reply:
Det har du rätt i. Kanske lägga till att det beror på vem man programmerar med

Reply:
detta problem har ju inte enbart med juniorer att göra, eller ens med parprogrammering - om man har en implicit eller explicit "policy", t.ex. "det ska finnas tester" så är det ju inte lämpligt om två personer approvar varandras policy-brytande PRs oavsett om de satt tillsammans eller ej. I WebClient-projektet så approvar vi andra som varit med enbart om det är en förhållandevis "trivial" PR - om det rör om i grytan eller är bra om fler känner till ändringarna, så låter vi den granskas av någon utanför mobben

Reply:
Jag la till en klausul om att det beror på :)